lib/httpclient/session.rb

```diff
@@ -942,19 +924,53 @@ def read_body_rest
           end
         end
         if buf && buf.bytesize > 0
-          yield buf
+          io.write(buf)
         else
           return
         end
       end
     end
 
+    def cast_to_io(to, block)
+      if to.respond_to?(:write)
+        yield to
+      elsif to
+        File.open(to, 'w+') do |file|
+          yield file
+        end
+      else
+        yield IOBlockAdapter.new(block)
+      end
+    end
+
+    class IOInflater
+      def initialize(io, inflater)
+        @io = io
+        @inflater = inflater
+      end
+
+      def write(chunk)
```

Comment:
Something that is not very well documented, is that `copy_steam` do accept fake IO objects as long as they respond to `#write()`. However it's kind of a fallback codepath, as it won't be able to use `sendfile()` so there will be no speed up.